### PR TITLE
Remove deprecated warning in TestMatrix, and reduce severity of identity server error

### PR DESCRIPTION
**Pull request by michaelkaye**
## Type of change

- [ ] Feature
- [ ] Bugfix
- [ ] Technical
- [X] Other : logging tidy-up

## Content

This exception is printed in the logs on every sign-out that did not have an identity server configured:
```
2022-03-08 10:02:48.129 14210-14267/? I/System.out: W/DefaultSignOutTask: Unable to disconnect identity server
2022-03-08 10:02:48.129 14210-14267/? I/System.out: org.matrix.android.sdk.api.session.identity.IdentityServiceError$NoIdentityServerConfigured
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.internal.session.identity.DefaultIdentityDisconnectTask.execute(IdentityDisconnectTask.kt:36)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.internal.session.identity.DefaultIdentityDisconnectTask.execute(IdentityDisconnectTask.kt:29)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.internal.session.signout.DefaultSignOutTask.execute(SignOutTask.kt:67)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.internal.session.signout.DefaultSignOutTask$execute$1.invokeSuspend(Unknown Source:15)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at kotlin.coroutines.jvm.internal.BaseContinuationImpl.resumeWith(ContinuationImpl.kt:33)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at kotlinx.coroutines.DispatchedTask.run(DispatchedTask.kt:106)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at kotlinx.coroutines.EventLoopImplBase.processNextEvent(EventLoop.common.kt:274)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at kotlinx.coroutines.BlockingCoroutine.joinBlocking(Builders.kt:85)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking(Builders.kt:59)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at kotlinx.coroutines.BuildersKt.runBlocking(Unknown Source:1)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at kotlinx.coroutines.BuildersKt__BuildersKt.runBlocking$default(Builders.kt:38)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at kotlinx.coroutines.BuildersKt.runBlocking$default(Unknown Source:1)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.common.CommonTestHelper.runBlockingTest(CommonTestHelper.kt:400)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.common.CommonTestHelper.signOutAndClose(CommonTestHelper.kt:433)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.common.CryptoTestData.cleanUp(CryptoTestData.kt:35)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.internal.crypto.verification.qrcode.VerificationTest.doTest(VerificationTest.kt:251)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.matrix.android.sdk.internal.crypto.verification.qrcode.VerificationTest.test_aliceAndBob_all_all(VerificationTest.kt:143)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at java.lang.reflect.Method.invoke(Native Method)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at net.lachlanmckee.timberjunit.TimberTestRule$TimberStatement.evaluate(TimberTestRule.java:185)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.rules.RunRules.evaluate(RunRules.java:20)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at androidx.test.ext.junit.runners.AndroidJUnit4.run(AndroidJUnit4.java:162)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.Suite.runChild(Suite.java:128)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.Suite.runChild(Suite.java:27)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
2022-03-08 10:02:48.129 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
2022-03-08 10:02:48.130 14210-14267/? I/System.out:     at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
2022-03-08 10:02:48.130 14210-14267/? I/System.out:     at org.junit.runner.JUnitCore.run(JUnitCore.java:137)
2022-03-08 10:02:48.130 14210-14267/? I/System.out:     at org.junit.runner.JUnitCore.run(JUnitCore.java:115)
2022-03-08 10:02:48.130 14210-14267/? I/System.out:     at androidx.test.internal.runner.TestExecutor.execute(TestExecutor.java:56)
2022-03-08 10:02:48.130 14210-14267/? I/System.out:     at androidx.test.runner.AndroidJUnitRunner.onStart(AndroidJUnitRunner.java:444)
2022-03-08 10:02:48.130 14210-14267/? I/System.out:     at android.app.Instrumentation$InstrumentationThread.run(Instrumentation.java:2145)
2022-03-08 10:02:48.130 14210-14267/? I/System.out: D/DefaultSignOutTask: SignOut: cleanup session...
```
This makes it look like a bad thing has happened, but practically in the case of no configured identity server, there's no problem.

I have also removed the usage of `MatrixConfiguration.Provider`, as I do not believe we use this piece of logic in the test suite at present.

## Motivation and context

I'm reading the logs a lot recently, and I keep finding this exception then having to discount it as not a real issue in the test suite.

By removing usage of `MatrixConfiguration.Provider`, we can remove `-PallWarningsAsError=false` and pick up new failures (and I also remove warnings from test runs in the development environment)

## Tests

I ran the test suite again and the logout exception wasn't printed in most cases, and the exception that would have been generated if we hadn't initialized the provider was also not thrown.

## Tested devices

- [ ] Physical
- [X] Emulator
- OS version(s):

## Checklist

- [ ] Changes has been tested on an Android device or Android emulator with API 21
- [ ] UI change has been tested on both light and dark themes
- [ ] Accessibility has been taken into account. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#accessibility
- [X] Pull request is based on the develop branch
- [ ] Pull request includes a new file under ./changelog.d. See https://github.com/vector-im/element-android/blob/develop/CONTRIBUTING.md#changelog
- [ ] Pull request includes screenshots or videos if containing UI changes
- [ ] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
- [ ] You've made a self review of your PR
- [ ] If you have modified the screen flow, or added new screens to the application, you have updated the test [UiAllScreensSanityTest.allScreensTest()](https://github.com/vector-im/element-android/blob/main/vector/src/androidTest/java/im/vector/app/ui/UiAllScreensSanityTest.kt#L73)

### CODE REVIEW

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/TestMatrix.kt [hunk 1]
```diff
@@ -105,16 +105,9 @@ internal class TestMatrix constructor(context: Context, matrixConfiguration: Mat
...
-            if (isInit.compareAndSet(false, true)) {
-                val appContext = context.applicationContext
-                if (appContext is MatrixConfiguration.Provider) {
-                    val matrixConfiguration = (appContext as MatrixConfiguration.Provider).providesMatrixConfiguration()
-                    instance = TestMatrix(appContext, matrixConfiguration)
-                } else {
-                    throw IllegalStateException("Matrix is not initialized properly." +
-                            " You should call Matrix.initialize or let your application implements MatrixConfiguration.Provider.")
-                }
+        fun getInstance(): TestMatrix {
```
Comment (ouchadam) [inline matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/TestMatrix.kt:108]: might be worth deleting these in favour of syncing with the production Matrix creation and only providing a `createInstance`, that's my bad for not updating this area as well https://github.com/vector-im/element-android/blob/develop/matrix-sdk-android%2Fsrc%2Fmain%2Fjava%2Forg%2Fmatrix%2Fandroid%2Fsdk%2Fapi%2FMatrix.kt#L106

Comment (michaelkaye) [reply matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/TestMatrix.kt:108]: I had a look at this; the problem I ran into was that we can only initialize it once; but the CommonTestHelper is created multiple times, so creates multiple TestMatrix  instances - so we'd need somewhere to store a static instance of it across all tests, so keeping it in TestMatrix seemed reasonable - in that sense the TestMatrix class is the thing that manages having a single instance of itself.

(I might be missing tricks though)

**Comment by michaelkaye** [comment]
Running nightly tests in https://github.com/vector-im/element-android/actions/runs/1951138219

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/TestMatrix.kt (hunk 1 above)
Comment (ouchadam) [reply matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/TestMatrix.kt:108]: I haven't dived too deep into the integration tests and this probably deserves an entirely separate discussion/issue :sweat_smile:  but I would have expected our test instance to not use a singleton state

**Review by ouchadam — APPROVED** [review]
:+1: for the original intention behind the PR

File: matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/TestMatrix.kt (hunk 1 above)
Comment (michaelkaye) [reply matrix-sdk-android/src/androidTest/java/org/matrix/android/sdk/common/TestMatrix.kt:108]: So the error came from creating an additional `androidx.work.WorkManager` on each initialization of TestMatrix. So I'm not sure quite whether we want to share one of those or if we want to tear it down on test finishing or ... 

TL;DR: this could be fixed but I don't want to dive into it to remove the errors as warnings flag.

**Comment by michaelkaye** [comment]
> Running nightly tests in https://github.com/vector-im/element-android/actions/runs/1951138219

Nightly tests are no more broken than normal - the flaky verification test and the UI test failure are expected.
